### PR TITLE
Add auth CLI

### LINKED ISSUE #565: Implement CLI support for auth (`dwave auth` subcommand)
Add `auth` subcommand to `dwave` CLI.

Add the following subcommands to `dwave auth`:
- `login` - initiate the OAuth flow. Use localhost-bound redirect URL, but support `--oob` (manual `code` & `state` entry)
- ~`list` - list existing identities/tokens~
- `get` - print access/refresh/id token
- `refresh` - refresh access/id token
- ~`revoke` - revoke a token~ (API not available)
- ~`ensure-valid` - refresh token only if needed~

### PR DESCRIPTION
**Pull request by randomir**
Close #565.

### CODE REVIEW

File: dwave/cloud/cli.py [hunk 1]
```diff
@@ -292,6 +323,18 @@ def _config_create(config_file, profile, ask_full=False):
     if profile != default_section and not config.has_section(profile):
         config.add_section(profile)
 
+    # set using Leap API
+    try:
+        sapi_token = _fetch_sapi_token(config_file, profile) if auto_token else None
+    except Exception as error:
+        click.echo(f"Failed to fetch SAPI token from Leap API ({error!s}).")
```
Comment (JoelPasvolsky) [inline dwave/cloud/cli.py:330]: Probably not this line but if I run `dwave auth login --profile test` and click `Decline` I get the error message `The provided authorization grant or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client`, which is incorrect. Corner case that can be an issue for later, not in the least bit urgent.

File: dwave/cloud/cli.py [hunk 2]
```diff
@@ -896,3 +943,156 @@ def setup(install_all, ask_full, verbose):
...
+    if not project:
+        raise CLIError(f'Project with {project_slug!r} ID, name or code not found. '
+                       'Please run "dwave leap project ls" to list available projects.', code=101)
+
+    # get token
+    project = project[0]
+    token = account.get_project_token(project=project)
+
+    output(f"Solver API token for project {project.name} ({project.code}) is {token}.",
+           token=token)
```
Comment (JoelPasvolsky) [inline dwave/cloud/cli.py:1098]: I get an error for one particular project which is (1) last in the list (2) contains another project name but different case:

```
dwave leap project ls --profile test
Leap projects:
 - ... 
 - jp_test2 ...
 - tonytest  ...
 - TonyTestbeta ...
```

`dwave leap project token --project "tonytest" --profile=test` works but

```
dwave leap project token --project "TonyTestbeta" --profile=test
Unhandled error: 1 validation error for _LeapProjectTokenResponse
data.token
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.4/v/string_type
```

Comment (randomir) [reply dwave/cloud/cli.py:1098]: This is a validation error of the Leap API response -- the token for that project is `None`, which according to specs is incorrect. Now, if `None` is in fact allowed value for token returned from Leap API, I'll update expected response specs.

/cc @charleswhchan

File: dwave/cloud/cli.py (hunk 1 above)
Comment (randomir) [reply dwave/cloud/cli.py:330]: Good catch.

File: dwave/cloud/cli.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/cli.py:1098]: Charles says this is a corner case for very old projects... I'll extend the model to support `null` tokens.

**Review by arcondello — APPROVED** [review]
LGTM

File: dwave/cloud/cli.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/cli.py:1102]: @JoelPasvolsky, this should now be fixed, feel free to test it again.

**Pull request merged**